### ISSUE TITLE
[BUG] Possible regression in Azure.Resource.AllowedRegions

### ISSUE THREAD

**Issue by mderriey** [Azure/PSRule.Rules.Azure#2461]
### Existing rule

Azure.Resource.AllowedRegions

### Description of the issue

The rule fails even though we haven't configured allowed locations with `Azure_AllowedRegions` or the new `AZURE_RESOURCE_ALLOWED_LOCATIONS` configuration settings.

This started happening when PSRule installed PSRule.Rules.Azure v1.30, where it was using v1.29 before.

### Error messages

```
 -> 66c6950b43a62 : Microsoft.Web/sites [9/10]

    [PASS] Azure.Resource.UseTags (AZR-000166)
    [FAIL] Azure.Resource.AllowedRegions (AZR-000167)
    | Template: templates/function.bicep:193:5

    | RECOMMEND:
    | Consider deploying resources to allowed regions to align with your
    | organizational requirements. Also consider using Azure Policy to enforce
    | allowed regions.

    | REASON:
    | - The location 'eastus' is not in the allowed set of resource locations.

    | HELP:
    | - https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Resource.AllowedRegions/

##[error]AZR-000167: 66c6950b43a62 failed Azure.Resource.AllowedRegions. Resources should be deployed to allowed regions.
```

### Reproduction

Here's our config file:

```yaml
binding:
  preferTargetInfo: true
  targetType:
    - type
    - resourceType

# Require minimum versions of modules.
requires:
  PSRule: '@pre >=2.9.0'
  PSRule.Rules.Azure: '@pre >=1.28.0'

# Use PSRule for Azure.
include:
  module:
    - PSRule.Rules.Azure

output:
  culture:
    - 'en-AU'

execution:
  unprocessedObject: Ignore

configuration:
  # Enable Bicep CLI checks.
  AZURE_BICEP_CHECK_TOOL: true

  # Enable automatic expansion of Azure parameter files.
  AZURE_PARAMETER_FILE_EXPANSION: true

  # Enable automatic expansion of Azure Bicep source files.
  AZURE_BICEP_FILE_EXPANSION: true

  # Configures the number of seconds to wait for build Bicep files.
  AZURE_BICEP_FILE_EXPANSION_TIMEOUT: 10

rule:
  # Enable custom rules that don't exist in the baseline
  includeLocal: true
```

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.30.0

**Comment by BernieWhite** [comment]
@mderriey Thanks for reporting the issue.

**Comment by mderriey** [comment]
Thanks @BernieWhite, that worked a treat 👌
